### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
### **Added:**
- [ ] Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- [ ] It will return the first user from the database.
- [ ] Create forms to perform the following functions:

> - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
> - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
> - Allow Users to add likes to Posts.

To get a clearer idea of the required functionalities, you can refer to the video demonstration provided in the link below.

https://github.com/Fahim-Khan-P/blog/assets/74261465/15f54db9-0151-44a7-a71e-ba3bfd8bc566

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    return unless @post.save
+
+    redirect_to user_posts_path
+  end
+
+  def new
+    @post = Post.new
+  end
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [ ] @Fahim-Khan-P great work on your post controller, however it's ruby convention that the new method comes before the create method, kindly interchange their order.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    return unless @post.save
+
+    redirect_to user_posts_path
+  end
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [ ] @Fahim-Khan-P Greate work on your create method, however, it's good practice to have a redirect route for both success and failure when creating a post, currently, it's only for when a post is created.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.post = Post.find(params[:post_id])
+    @comment.author_id = current_user.id
+    redirect_to user_post_path(id: @comment.post_id) if @comment.save
+  end
```
Comment (danmainah) [inline app/controllers/comments_controller.rb]: - [ ] @Fahim-Khan-P Great work on your create method, however it's good practice to have a redirect path for when a comment is created and when it fails, currently you only have a path for when its created.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    @post = Post.new
+  end
+
+  def create_comment
+    @comment = Comment.new(comment_params)
+    @comment.post = Post.find(params[:id])
+    @comment.author_id = current_user.id
+    redirect_to user_post_path if @comment.save
+  end
+
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [ ] @Fahim-Khan-P Kindly remove this method, its good practice to ensure the controller only has methods for a particular view, you already have this method in the comments controller

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Fahim-Khan-P , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Added all forms required to add a post, comment and like.
✔️ Professional written README file with all sections required.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly ensure to not add any commits after asking for a review, always wait until the review is submitted.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Fahim-Khan-P) [reply app/controllers/posts_controller.rb]: Thank you for your feedback! I agree with your suggestion to follow the Ruby convention and have updated the order of the new and create methods in the PostsController accordingly. Thank you for bringing this to my attention.

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Fahim-Khan-P) [reply app/controllers/posts_controller.rb]: Thank you for your feedback! I agree that it's good practice to have a redirect route for both success and failure when creating a post. I have updated the create method in the PostsController to handle both cases.

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (Fahim-Khan-P) [reply app/controllers/comments_controller.rb]: Thank you for your feedback! I agree that it's good practice to have a redirect route for both success and failure when creating a comment. I have updated the create method in the **CommentsController** to handle both cases:

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (Fahim-Khan-P) [reply app/controllers/posts_controller.rb]: Thank you for your feedback! I agree that it's a duplicated block, I have removed it.

**Review by uwadonat — APPROVED** [review]
## Hi @Fahim-Khan-P,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Detailed pull request message with video ✔️
- Correct workflow was used ✔️
- No linter errors ✔️
- Professional commit messages ✔️
thanks-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Fahim-Khan-P** [comment]
@uwadonat,
Bundle of thanks!

**Pull request merged**